chest.lua

```diff
@@ -11,35 +11,69 @@ local function get_item_count(pos, player, count)
 	return count
 end
 
+-- Just in case we are under MineClone
+local default = rawget(_G, "default") or nil
+if default == nil then
+	default = rawget(_G, "mcl_sounds")
+end
+
 local tex_mod = "^[colorize:#FF2:50"
 minetest.register_node("simple_protection:chest", {
 	description = S("Shared Chest") .. " " .. S("(by protection)"),
 	tiles = {
-		"default_chest_top.png"  .. tex_mod,
-		"default_chest_top.png"  .. tex_mod,
-		"default_chest_side.png" .. tex_mod,
-		"default_chest_side.png" .. tex_mod,
-		"default_chest_side.png" .. tex_mod,
-		"default_chest_lock.png" .. tex_mod
+		"simple_protection_chest_top.png"  .. tex_mod,
+		"simple_protection_chest_top.png"  .. tex_mod,
+		"simple_protection_chest_side.png" .. tex_mod,
+		"simple_protection_chest_side.png" .. tex_mod,
+		"simple_protection_chest_side.png" .. tex_mod,
+		"simple_protection_chest_lock.png" .. tex_mod
 	},
 	paramtype2 = "facedir",
 	sounds = default.node_sound_wood_defaults(),
-	groups = {choppy = 2, oddly_breakable_by_hand = 2},
+	groups = {
+		-- Minetest Game
+		choppy = 2, oddly_breakable_by_hand = 2,
+		-- MineClone
+		handy=1,axey=1, deco_block=1,
+	},
+	_mcl_blast_resistance = 2.5,
+	_mcl_hardness = 2.5,
 
 	after_place_node = function(pos, placer)
 		local meta = minetest.get_meta(pos)
 		meta:set_string("infotext", S("Shared Chest"))
-		meta:set_string("formspec",
-			"size[8,9]" ..
-			default.gui_bg ..
-			default.gui_bg_img ..
-			"list[context;main;0,0.3;8,4;]" ..
-			"list[current_player;main;0,5;8,4;]" ..
-			"listring[context;main]" ..
-			"listring[current_player;main]"
-		)
-		local inv = meta:get_inventory()
-		inv:set_size("main", 8*4)
+		if minetest.registered_items["default:dirt"] then
+			meta:set_string("formspec",
+				"size[8,9]" ..
+				default.gui_bg ..
+				default.gui_bg_img ..
+				"list[context;main;0,0.3;8,4;]" ..
+				"list[current_player;main;0,5;8,4;]" ..
+				"listring[context;main]" ..
+				"listring[current_player;main]"
+			)
+			local inv = meta:get_inventory()
+			inv:set_size("main", 8*4)
+		elseif minetest.registered_items["mcl_core:dirt"] then
+			-- In MineClone they have a separate overlay of images for the inventory,
+			-- This branch should setup the proper dimentions (9x3) instead of Minetest's fairly nice dimentions (8x4)
+			local mcl_formspec = rawget(_G, "mcl_formspec") or nil
```

Comment:
Access `mcl_formspec` directly and whitelist the variable in `.luacheckrc`. Things would error anyway if this were missing in MineClone 2.